### ISSUE TITLE
Coupling to HTTP

### ISSUE THREAD

**Issue by rubensworks** [RubenVerborgh/solid-server-architecture#39]
This is an issue to make sure that we keep in mind that this server could be extended to support non-HTTP protocols, such as IPFS.
As far as I can see, the LdpHandler is the only component that has a link to the HTTP layer.
So architecturally, I think we're good, unless I'm missing something.

Implementation-wise, it will be important that this LDP-HTTP link goes through some kind of dependency injection layer to allow pluggin in custom protocols (such as IPFS).

Related to #24.

**Comment by RubenVerborgh** [comment]
Good point; I have indeed minimized the dependency on HTTP to the extent I saw possible. The main driver was there to have a layered system, and specifically to go against the hard-to-test HTTP-handler-driven implementation of node-solid-server.

Concretely, changes would remain limited to:
- The `HTTP`, which would be replaced.
- The extractors/parsers/writers in the `LDP` and `Authentication` packages, which are protocol-specific.

Do we need any action on this?
